pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -199,6 +208,22 @@ func (c *ClusterConfig) ValidateVPCConfig() error {
 	return nil
 }
 
+func (c *ClusterConfig) ipv6CidrsValid() error {
+	count := 0
+	if c.VPC.IPv6Cidr != "" {
+		count++
+	}
+
+	if c.VPC.IPv6Pool != "" {
+		count++
+	}
+
+	if count == 0 || count == 2 {
+		return nil
+	}
+	return fmt.Errorf("Ipv6Cidr and Ipv6Pool must both be configured to use a custom IPv6 CIDR and address pool")
+}
+
```

Comment:
I don't get this function. :D You aren't saying `"Ipv6Cidr and Ipv6Pool must both be configured to use a custom IPv6 CIDR and address pool` but you are allowing for them to be not configured at all on that count == 0 or.

So is it now that they need to be configured if either is defined or not? Because if they need to be configured than the count is not needed at all.

Also, I find that reading this with the counting increases the cognitive load of this function significantly mostly because it's unexpected format. I would rather see a logical match like:

```go
func (c *ClusterConfig) ipv6CidrsValid() error {
	if (c.VPC.IPv6Cidr == "" && c.VPC.IPv6Pool == "") || (c.VPC.IPv6Cidr != "" && c.VPC.IPv6Pool != "") {
		return nil
	}
	return fmt.Errorf("Ipv6Cidr and Ipv6Pool must both be configured to use a custom IPv6 CIDR and address pool")
}
```

which just reads a lot smoother. WDYT?